### DISCUSSION TITLE
[RFCS FS-1114] Discussion for ML compatibility revisions

### DISCUSSION THREAD

**Discussion by dsyme** [fsharp/fslang-design#626] [Language and core library RFC discussions]
Discussion for RFC FS-1114 https://github.com/fsharp/fslang-design/blob/main/RFCs/FS-1114-ml-compat-revisions.md

**Comment by kerams** [comment]
If these have been marked as deprecated for so long, why not outright purge the features from the language? I imagine it could simplify the parser quite a bit.

**Reply by dsyme** [reply]
Microsoft have product support and backwards compat obligations

**Reply by dsyme** [reply]
> I imagine it could simplify the parser quite a bit.

We will likely duplicate the parser into a "backwards compat" parser (`/langversion:5.0` and before) and a simpler "indentation-aware-only parser", just for the simplifying benefits.

**Comment by kryptine** [comment]
Doesn't [F*](https://github.com/FStarLang/FStar) make use of this feature in order to compile on both F# and OCaml?

**Reply by smoothdeveloper** [reply]
AFAIR, from engaging on some discussions with F* contributors, they have been heading towards severing from F#, although it is still a bootstrapping requirement if OCaml is not available, on the old dotnet/mono toolchain.

Also, AFAIU, the subset of F# used in the bootstrapping of F* doesn't need to leverage last F# advancements, for the purpose of bootstrapping.

In this context, I doubt it would impact the project significantly that the deprecation is taking place in newer versions of the compiler.
